### DISCUSSION TITLE
Project referencing with resources

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#17323] [General]
This may be a vs tooling question but will try here first.

I have a Uno.Sdk project, which has 2 references to other uno.sdk class libraries.
One reference called common contains the resourcedictionary.xaml that will contain all the UI resource setting i wish for this project.
In my main Uno.Sdk project i have the App.xaml make the reference to the common class library and its resources like this.
`<Application.Resources>

  <ResourceDictionary>

    <ResourceDictionary.MergedDictionaries>

      

      <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />

      

      <ToolkitResources xmlns="using:Uno.Toolkit.UI" />

      

      <ResourceDictionary Source="ms-appx:///Common/AppResources.xaml" />

    </ResourceDictionary.MergedDictionaries>

  </ResourceDictionary>

</Application.Resources>`

Now i have another Uno.Sdk class library that I want to contain the screens that will be used in the app, lets call it views

The question is how can i have the views class reference the resources in common?  And at design time how can the vs editor know about these static resources?

thanks

**Comment by jeromelaban** [comment]
Thanks for the question. You can find a guide on how to [create control libraries](https://platform.uno/docs/articles/guides/how-to-create-control-libraries.html), which is similar to creating common resources.

The Visual Studio XAML intellisense will properly detect resources referenced from app.xaml, even if they're included in a separate class library.
